scripts/generate_test_suites.py

```diff
@@ -0,0 +1,44 @@
+import os
+import sys
+
+parts = sys.argv[1]
+
+instrumentation_tests_path = 'instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests'
+
+
+def file_has_tests(path, file):
+	file_content = open(f'{path}/{file}').read()
+	return '@Test' in file_content
+
+
+def get_file_package(path):
+	inner_path_part = path.replace(instrumentation_tests_path, '').replace('/', '.')
+	return f'com.mapbox.navigation.instrumentation_tests{inner_path_part}'
+
+
+test_matrix = {}
+for root, dirs, files in os.walk(f'{instrumentation_tests_path}'):
+	for file in files:
+		if file_has_tests(root, file):
+			test_matrix[f'{file}'] = get_file_package(root)
+
+suite_size = len(test_matrix.keys()) // int(parts) + 1
+suites = {}
+for index, test in enumerate(test_matrix.keys()):
+	part_number = int(index / suite_size) + 1
+	if part_number in suites:
+		suites[part_number].append(test)
```

Comment:
NIT, not for the first iteration.
Ideally we'd split tests based on duration. And actually it's common that consecutive tests (for example, tests in 1 class), have approximately the same duration. For instance, when we test route refresh, there are some timeouts in play and we have to wait for some time. The same thing with alternatives: alternatives won't appear fast, we'd probably need to travel along the route for quite some time before we can actually test anything.
So while the current approach does improve the situation a lot, there are still many things that we should take into consideration to keep improving.

Reply:
correct, this way divide just by classes. In this way we can face the problem where 7 classes have 7 tests and another 7 classes have 50 tests. I'll implement more smart divider later.